smack-omemo/src/main/java/org/jivesoftware/smackx/omemo/OmemoManager.java

```diff
@@ -289,6 +289,9 @@ public static void addExplicitMessageEncryptionHint(Message omemoMessage) {
      * @throws SmackException.NoResponseException   wrong
      */
     public boolean resourceSupportsOmemo(Jid resource) throws XMPPException.XMPPErrorException, SmackException.NotConnectedException, InterruptedException, SmackException.NoResponseException {
```

Comment:
This should probably be a FullJid then. And I'd name it fullJid, not 'resource'.